fhir/src/main/java/gov/cms/ab2d/fhir/IdentifierUtils.java

```diff
@@ -100,32 +161,113 @@ private static String getValue(ICompositeType identifier) {
         return (String) Versions.invokeGetMethod(identifier, "getValue");
     }
 
-    private static boolean isBeneficiaryId(ICompositeType identifier) {
-        String system = getSystem(identifier);
-        String value = getValue(identifier);
-        if (StringUtils.isAnyBlank(system, value)) {
-            return false;
+    /**
+     * From a FHIR identifier include MBI information, get the currency information
+     *
+     * @param identifier - the identifier
+     * @return - the currency information
+     */
+    private static PatientIdentifier.Currency getCurrencyMbi(ICompositeType identifier) {
+        Optional<IBaseExtension> currencyExtension = getCurrencyExtension(identifier);
+        if (currencyExtension.isEmpty()) {
+            return PatientIdentifier.Currency.UNKNOWN;
+        }
+        Object code = Versions.invokeGetMethod(currencyExtension.get(), "getValue");
+        String codeValue = (String) Versions.invokeGetMethod(code, "getCode");
+        switch (codeValue) {
```

Comment:
A little clunky, better yet to have a static helper method within the Currency enum that converts a codeValue to an instance.  The translation could happen either through a static map *or* having 'codeValue' be a property on the enum and a search through all enums.

In any case, too much detail here that would benefit from being abstracted away.